spikeextractors/extractors/neuroscopesortingextractor/neuroscopesortingextractor.py

```diff
@@ -26,17 +37,31 @@ class NeuroscopeSortingExtractor(SortingExtractor):
     is_writable = True
     mode = 'custom'
 
-    def __init__(self, resfile, clufile):
+    def __init__(self, resfile, clufile, keep_mua_units=True):
         SortingExtractor.__init__(self)
         res = np.loadtxt(resfile, dtype=np.int64, usecols=0, ndmin=1)
         clu = np.loadtxt(clufile, dtype=np.int64, usecols=0, ndmin=1)
         if len(res) > 0:
+            # Extract the number of clusters read as the first line of the clufile
+            # then remove it from the clu list
             n_clu = clu[0]
             clu = np.delete(clu, 0)
+            
+            # Initialize spiketrains and extract times from .res and appropriate clusters from .clu
+            # based on user input for ignoring multi-unit activity
             self._spiketrains = []
-            self._unit_ids = list(x + 1 for x in range(n_clu))
-            for s_id in self._unit_ids:
-                self._spiketrains.append(res[(clu == s_id).nonzero()])
+            if keep_mua_units: # default
+                n_clu -= 1;
+                self._unit_ids = list(x+1 for x in range(n_clu)) # generates list from 1,...,clu[0]-1
+                for s_id in self._unit_ids:
+                    self._spiketrains.append(res[(clu == s_id).nonzero()])
+            else:
+                # Ignoring IDs of 0 until get_unsorted_spike_train is implemented into base
+                # Also ignoring IDs of 1 since user called keep_mua_units=False
+                n_clu -= 2;
+                self._unit_ids = list(x+1 for x in range(n_clu)) # generates list from 1,...,clu[0]-2
+                for s_id in self._unit_ids:
+                    self._spiketrains.append(res[(clu == s_id+1).nonzero()]) # only reading cluster IDs 2,...,clu[0]-1
         else:
             self._spiketrains = []
             self._unit_ids = []
```

Comment:
Can you add the new argument in the `self._kwargs`?

```
self._kwargs = {'resfile': str(Path(resfile).absolute()),
                         'clufile': str(Path(clufile).absolute()),
                          'keep_mua_units': keep_mua_units
}
```

This is done in order to make the extractors dumpable, so they can be re-instantiated just from their kwargs

Reply:
Done.